### PR TITLE
feat(chart): Replace upstream Dex with Bitnami image

### PR DESCRIPTION
**Pull request by javirln**
This patch replaces the upstream Chart of Dex in favor of custom manifests using Bitnami Dex image.

#1151 https://github.com/bitnami/charts/pull/27100

### CODE REVIEW

**Comment by javirln** [comment]
This PR needs to be merged first: https://github.com/chainloop-dev/chainloop/pull/1181

File: deployment/chainloop/values.yaml [hunk 1]
```diff
@@ -1080,45 +1103,532 @@ vault:
...
+  ## @param dex.image.digest Dex image digest in the way sha256:aa.... Please note this parameter, if set, will override the tag
+  ## @param dex.image.pullPolicy Dex image pull policy
+  ## @param dex.image.pullSecrets Dex image pull secrets
+  ## @param dex.image.debug Enable Dex image debug mode
+  ##
+  image:
+    registry: docker.io
+    repository: bitnami/dex
+    tag: 2.40.0-debian-12-r1
+    digest: ""
```
Comment (migmartri) [inline deployment/chainloop/values.yaml:1125]: you can remove this

File: deployment/chainloop/templates/dex/config.configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{{- /*
+Copyright Chainloop, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+apiVersion: v1
+kind: ConfigMap
```
Comment (migmartri) [inline deployment/chainloop/templates/dex/config.configmap.yaml:7]: move to secret

File: deployment/chainloop/templates/dex/config.configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+metadata:
+  name: {{ include "chainloop.dex.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: dex
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+data:
```
Comment (migmartri) [inline deployment/chainloop/templates/dex/config.configmap.yaml:16]: stringData

File: deployment/chainloop/templates/dex/config.configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          {{ $controlPlaneUrl := include "chainloop.controlplane.external_url" . }}
+          {{- if not (eq $controlPlaneUrl "null") -}}
+          - "{{ $controlPlaneUrl }}/auth/callback"
+          {{- end -}}
+        secret: ZXhhbXBsZS1hcHAtc2VjcmV0
+
+    # required to enable static passwords
+    enablePasswordDB: true
+
+    staticPasswords:
```
Comment (migmartri) [inline deployment/chainloop/templates/dex/config.configmap.yaml:41]: move to values.yaml

File: deployment/chainloop/templates/dex/config.configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+{{- /*
+Copyright Chainloop, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+apiVersion: v1
```
Comment (migmartri) [inline deployment/chainloop/templates/dex/config.configmap.yaml:6]: make sure it happens in dev only

**Review by migmartri — COMMENTED** [review]
I'd follow this with moving things to a subchart

**Review by migmartri — APPROVED** [review]

**Pull request merged**